framework/include/messaging/messaging.h

```diff
@@ -0,0 +1,159 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/**
+ * @defgroup Messaging Messaging
+ * @ingroup Messaging
+ * @brief Provides APIs for Messaging
+ * @{
+ */
+
+/**
+ * @file messaging/messaging.h
+ */
+#ifndef __MESSAGING_H__
+#define __MESSAGING_H__
+
+/**
+ * @brief Called when a message is received
+ */
+typedef void (*msg_callback_t)(void *callback_data, void *user_data);
+
+/**
+ * @brief The type of sending message
+ * @details MSG_SEND_TYPE_SYNC  : A sender waits after sending message until receiving a reply message.\n
+ *	    MSG_SEND_TYPE_ASYNC : A sender continues to run after sending message.
+ */
+enum msg_send_type_e {
+	MSG_SEND_TYPE_SYNC = 1,
+	MSG_SEND_TYPE_ASYNC = 2,
+	MSG_SEND_TYPE_MAX
+};
+typedef enum msg_send_type_e msg_send_type_t;
+
+/**
+ * @brief The structure of parameters which used for sending
+ */
+struct msg_send_param_s {
+	int priority;
+	msg_send_type_t type; /* for sender : MSG_SEND_TYPE_SYNC or MSG_SEND_TYPE_ASYNC */
+	const char *msg;
+	int msglen;
+};
+typedef struct msg_send_param_s msg_send_param_t;
+
+/**
+ * @brief The structure of parameters which used for replying
+ */
+struct msg_reply_param_s {
+	char *msg;
+	int msglen;
+};
+typedef struct msg_reply_param_s msg_reply_param_t;
```

Comment:
1. Is this param? or reply data?
2. It is same as msg_recv_data_s. Do we need to keep as different structure?